src/node_i18n.cc

```diff
@@ -62,6 +72,306 @@ bool flag_icu_data_dir = false;
+    return ret;
+  }
+  return Buffer::Copy(isolate, dst, len);
+}
+
+struct Converter {
+  Converter(Environment* env, const char* name)
+      : conv(nullptr) {
+    UErrorCode status = U_ZERO_ERROR;
+    conv = ucnv_open(name, &status);
+    CHECK(U_SUCCESS(status));
+  }
+
+  ~Converter() {
+    if (conv != nullptr)
+      ucnv_close(conv);
+  }
+
+  UConverter* conv;
+};
+
+// One-Shot Converters
+
+// Converts the Buffer from one named encoding to another.
+// args[0] is a string identifying the encoding we're converting to.
+// args[1] is a string identifying the encoding we're converting from.
+// args[2] must be a buffer instance
+void Convert(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  CHECK(args[0]->IsString());
+  CHECK(args[1]->IsString());
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[2]);
+
+  Utf8Value to_name(isolate, args[0]);
+  Utf8Value from_name(isolate, args[1]);
+
+  SPREAD_BUFFER_ARG(args[2], ts_obj);
+
+  UErrorCode status = U_ZERO_ERROR;
+
+  MaybeStackBuffer<char> buf;
+  uint32_t limit;
+  char* target;
+  const char* data = ts_obj_data;
+
+  Converter to(env, *to_name);
+  Converter from(env, *from_name);
+  if (to.conv == nullptr || from.conv == nullptr)
+    return;
+
+  ucnv_setSubstChars(to.conv, "?", 1, &status);
+  status = U_ZERO_ERROR;
+
+  limit = ts_obj_length * ucnv_getMaxCharSize(to.conv);
+  buf.AllocateSufficientStorage(ts_obj_length * ucnv_getMaxCharSize(to.conv));
+  target = *buf;
+
+  ucnv_convertEx(to.conv, from.conv,
+                 &target, target + limit,
+                 &data, ts_obj_data + ts_obj_length,
+                 nullptr, nullptr, nullptr, nullptr,
+                 true, true,
+                 &status);
+
+  if (U_SUCCESS(status)) {
+    MaybeLocal<Object> ret = AsBuffer(isolate, &buf, target - *buf);
+    if (!ret.IsEmpty())
+      return args.GetReturnValue().Set(ret.ToLocalChecked());
+  }
+
+  ThrowICUError(env, status, "Unable to transcode buffer");
+}
+
+// Converts to UCS2 from ISO-8859-1 and US-ASCII
+// args[0] is the encoding to convert from
+// args[1] must be a buffer instance
+void ConvertToUcs2(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  CHECK(args[0]->IsString());
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[1]);
+
+  Utf8Value name(isolate, args[0]);
+
+  SPREAD_BUFFER_ARG(args[1], ts_obj);
+
+  UErrorCode status = U_ZERO_ERROR;
+  size_t len;
+
+  MaybeStackBuffer<UChar> buf(ts_obj_length);
+
+  Converter from(env, *name);
+  if (from.conv == nullptr)
+    return;
+
+  len = ucnv_toUChars(from.conv,
+                      *buf, ts_obj_length * sizeof(*buf),
+                      ts_obj_data, ts_obj_length,
+                      &status);
+
+  if (U_SUCCESS(status)) {
+    MaybeLocal<Object> ret = AsBuffer(isolate, &buf, ts_obj_length << 1);
```

Comment:
sizeof.  You could and probably should DRY it with the code above and capture it in a `const size_t`.